packs/boundary_worker/templates/pack.nomad.tpl

```diff
@@ -0,0 +1,173 @@
+      provider = "[[ $service_provider ]]"
+
+      [[ if eq $port.omit_check false ]]
+      # see https://developer.hashicorp.com/nomad/docs/job-specification/check
+      check {
+        name     = "[[ $name ]]"
+        type     = "[[ $port.type ]]"
+        [[- if eq $port.type "http" ]]
+        method   = "[[ $port.method ]]"
+        path     = "[[ $port.path ]]"
+        [[- end ]]
+        interval = "[[ $port.check_interval ]]"
+        timeout  = "[[ $port.check_timeout ]]"
+      }
+      [[- end ]]
+    }
+    [[ end ]]
+
+    # see https://developer.hashicorp.com/nomad/docs/job-specification/restart
+    restart {
+      [[- $restart_logic := var "nomad_group_restart_logic" . ]]
+      attempts = [[ $restart_logic.attempts ]]
+      interval = "[[ $restart_logic.interval ]]"
+      delay    = "[[ $restart_logic.delay ]]"
+      mode     = "[[ $restart_logic.mode ]]"
+    }
+
+    # see https://developer.hashicorp.com/nomad/docs/job-specification/volume
+    [[/* iterate over `var.volumes` to create Volumes */]]
+    [[- range $index, $mount := var "nomad_group_volumes" . ]]
+    volume "[[ $mount.name ]]" {
+      source    = "[[ $mount.name ]]"
+      type      = "[[ $mount.type ]]"
+      read_only = [[ $mount.read_only ]]
+    }
+    [[ end ]]
+
+    # see https://developer.hashicorp.com/nomad/docs/job-specification/task
+    task "[[ var "nomad_task_name" . ]]" {
+      # see https://developer.hashicorp.com/nomad/docs/drivers
+      driver = "[[ var "nomad_task_driver" . ]]"
+
+      # Action to print environment variables relevant to the Boundary Helper
+      # see https://developer.hashicorp.com/nomad/docs/job-specification/action
+      action "print-env" {
+        command = "[[ var "app_boundary_helper_path" . ]]"
+
+        # see https://github.com/workloads/boundary-helper/#environment-variables
+        args = [
+          "env"
+        ]
+      }
+
+      # Action to clean up old workers based on "Last Seen" duration
+      # see https://developer.hashicorp.com/nomad/docs/job-specification/action
+      action "cleanup-workers" {
+        command = "[[ var "app_boundary_helper_path" . ]]"
+
+        # see https://github.com/workloads/boundary-helper/#environment-variables
+        args = [
+          "cleanup-workers"
+        ]
+      }
+
+      # see https://developer.hashicorp.com/nomad/docs/drivers/raw_exec
+      # and https://developer.hashicorp.com/nomad/docs/drivers/exec
+      config {
+        command = "boundary"
+
+        args = [
+          "server",
+          "-config",
+          "${NOMAD_TASK_DIR}/config.hcl"
+        ]
+      }
+
+        # set `NOMAD_TOKEN` using Workload Identity
+      # see https://developer.hashicorp.com/nomad/docs/concepts/workload-identity
+      identity {
+        env = true
+      }
+
+      [[ template "environment_variables" . ]]
+
+      # see https://developer.hashicorp.com/nomad/docs/job-specification/template
+      template {
+        change_mode = "restart"
```

Comment:
Some config changes can be handled by a reload, but I don't know which ones and if they're enough :D

Reply:
"restart" is the preferred option here. From the [Worker](https://developer.hashicorp.com/boundary/docs/configuration/worker#signals) docs:

> The SIGHUP signal causes a worker to reload its configuration file to pick up any updates for the `initial_upstreams` and `tags` values. Any other updated values are ignored.

For the primary use-case of this Pack, `initial_upstreams` isn't as important, and `tags` would primarily change if the rendered-and-run Pack would be re-allocated.